File: app/src/main/java/com/example/medplus/model/Notification.kt
package com.example.medplus.model

import com.google.firebase.Timestamp

/**
 * Data model for notifications in MedPlus.
 */
data class Notification(
    val id: String = "",
    val userId: String = "",
    val title: String = "",
    val message: String = "",
    val type: String = "GENERAL", // e.g., APPOINTMENT, VERIFICATION, SYSTEM
    val isRead: Boolean = false,
    val timestamp: Timestamp = Timestamp.now()
)

File: backend/models/Notification.js
const mongoose = require('mongoose');

const NotificationSchema = new mongoose.Schema({
  userId: { type: String, default: null }, // Null if it is an Admin Notification
  doctorId: { type: String, default: null }, // Relevant if notification is about a doctor (e.g. for admin)
  title: { type: String, required: true },
  message: { type: String, required: true },
  type: { type: String, default: 'GENERAL' }, // GENERAL, APPOINTMENT, VERIFICATION, SYSTEM, DOCTOR_VERIFICATION
  isRead: { type: Boolean, default: false },
  timestamp: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Notification', NotificationSchema);
